backend/applications/api/v1/serializers.py

```diff
@@ -0,0 +1,55 @@
+from rest_framework import serializers
+
+from applications.models import Application, SummerBill
+from applications.services import update_summer_bills_using_api_data
+from companies.api.v1.serializers import CompanySerializer
+
+
+class SummerBillSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SummerBill
+        fields = [
+            "id",
+            "summer_bill_id",
+            "contact_name",
+            "contact_email",
+            "work_postcode",
+            "employee_name",
+            "employee_school",
+            "employee_ssn",
+            "employee_phone_number",
+            "is_unnumbered_summer_bill",
+            "unnumbered_summer_bill_reason",
+        ]
+
+
+class ApplicationSerializer(serializers.ModelSerializer):
+    summer_bills = SummerBillSerializer(many=True, required=False, allow_null=True)
+
+    class Meta:
+        model = Application
+        fields = [
+            "id",
+            "status",
+            "invoicer_name",
+            "invoicer_email",
+            "invoicer_phone_number",
+            "summer_bills",
+        ]
+
+    def update(self, instance, validated_data):
+        request = self.context.get("request")
+        summer_bills_data = validated_data.pop("summer_bills", [])
+        if request and request.method == "PUT":
+            update_summer_bills_using_api_data(summer_bills_data, instance)
+
+        return super().update(instance, validated_data)
+
+    def to_representation(self, obj):
+        data = super().to_representation(obj)
+
+        request = self.context.get("request")
+        if request and request.method == "GET":
+            # Only show the related company on GET request since we do not want to update it manually.
```

Comment:
It might be just me but I don't quite understand what this comment is saying 😄

Reply:
I'll try to restructure it a bit :D

Basically what I mean is that `company` is read-only.

Reply:
Also I'm not sure if this is the correct way to implement this in the serializer, so that when reading applications, it would return the `company` but would not allow updating it.

Reply:
You could also add it as a read-only field. One benefit would be that it would be discovered by automatic documentation generators.

Reply:
Well that would be only reasonable :D Forgot that I can do that.